component/generated-rolebindings.jsonnet

```diff
@@ -38,6 +38,57 @@ local generateDefaultRolebindingInNsPolicy = kyverno.ClusterPolicy('default-role
           },
         },
       },
+      {
+        name: 'namespace-edit-role',
+        match: common.MatchOrgNamespaces,
+        generate: {
+          kind: 'Role',
+          synchronize: false,
+          name: params.generatedNamespaceOwnerRole.name,
+          namespace: '{{request.object.metadata.name}}',
+          data: {
+            rules: [
+              {
+                apiGroups: [
+                  '',
+                ],
+                resources: [
+                  'namespaces',
```

Comment:
Uhmm, I think this is fundamentally wrong? If we really allow Get on `Namespaces` without the `resourceName` property, users then can list ALL namespaces. That's not something we want. What we rather want is that users can edit `Projects`. Projects are "virtual" resources provided by OpenShift, with the difference that Openshift verifies and filters the Get, list etc. to only those where the user has access to.

Besides, I believe that OpenShift by default already allows edits/patches etc on Projects, so why the additional Roles/Rolebinding?

Or did I misunderstand something?

Reply:
Because we create a `Role` and `RoleBinding` in the namespace, we don't grant users cluster-scope permissions to get individual namespaces. Additionally, `get` and `list` are separate verbs, so listing namespaces doesn't work in any case:

```
$ oc whoami
jane.doe
$ kubectl get ns
Error from server (Forbidden): namespaces is forbidden: User "jane.doe" cannot list resource "namespaces" in API group "" at the cluster scope
$ kubectl get ns kube-system
Error from server (Forbidden): namespaces "kube-system" is forbidden: User "jane.doe" cannot get resource "namespaces" in API group "" in the namespace "kube-system"
$ kubectl get ns jane-doe-test1
NAME             STATUS   AGE
jane-doe-test1   Active   20d
$ kubectl get project
NAME             DISPLAY NAME   STATUS
jane-doe-test1   test1          Active
$ kubectl get projects
NAME             DISPLAY NAME   STATUS
jane-doe-test1   test1          Active
```

Also, since you can't edit/add/remove labels or annotations on `Project` resources (see below), we have to allow users to patch namespaces which they have access to.

```
$ kubectl annotate project jane-doe-test1 test=tester
The Project "jane-doe-test1" is invalid: metadata.annotations[test]: Invalid value: "tester": field is immutable, try updating the namespace
$ kubectl label project jane-doe-test1 test=tester
The Project "jane-doe-test1" is invalid: metadata.labels[test]: Invalid value: "tester": field is immutable, , try updating the namespace
```

Reply:
Ok. Thanks for the clarification.
I'll mark this as a "Friday" bug in my thinking :)